src/devices/src/virtio/net/tap.rs

```diff
@@ -193,6 +193,7 @@ impl Tap {
         if ret == -1 {
             return Err(IoError::last_os_error());
         }
+        #[allow(clippy::cast_sign_loss)] // writev only returns -1 or positive values
```

Comment:
In this case you can do `Ok(usize::try_from(ret).unwrap())` instead of `Ok(ret as usize)`.